src/librustc/ty/item_path.rs

```diff
@@ -76,6 +77,11 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         self.item_path_str(self.hir().local_def_id(id))
     }
 
+    // FIXME(@ljedrz): replace the NodeId variant
+    pub fn hir_path_str(self, id: hir::HirId) -> String {
```

Comment:
Please do not use this - #57967 (or the refactoring that it's based on, rather, will split it into a separate PR) removes `node_path_str`.

If you really need to replace `node_path_str` uses, you can call `item_path_str` directly (a lot of the time you might even have a `DefId` that was being converted to a `NodeId`!).

Reply:
I opened #58140 in the meanwhile, and it removes `hir_path_str`, so it's compatible with any PR also removing `hir_path_str`.